propagation.go

```diff
@@ -112,8 +112,27 @@ type TextMapReader interface {
 	ForeachKey(handler func(key, val string) error) error
 }
 
+// TextMapCarrier allows the use of regular map[string]string
+// as both TextMapWriter and TextMapReader.
```

Comment:
@bensigelman  I don't think we have multi-map *requirement* in the current API. The Writer doc says

```
	// Set a key:value pair to the carrier. Multiple calls to Set() for the
	// same key leads to undefined behavior.
```
so `map[string]string` satisfies that by overriding the key.

And the reader API says multiple keys *may* be returned, but map wouldn't do it.
```
	// NOTE: A single `key` may appear in multiple calls to `handler` for a
	// single `ForeachKey` invocation.
```